modules/record_collection_data/variables.tf

```diff
@@ -0,0 +1,29 @@
+variable "zone_id" {
+  description = "Cloudflare Hosted Zone ID"
+  type        = string
+}
+
+variable "name" {
+  description = "Cloudflare Hosted Zone Record Name"
+  type        = string
+}
+
+variable "type" {
+  description = "Cloudflare Hosted Zone Record Type"
+  type        = string
+}
+
+variable "data" {
```

Comment:
It should be list of objects if you have multiple data records with the same name

Reply:
> It should be list of objects if you have multiple data records with the same name

This will work as-is in such a case.